### ISSUE TITLE
'Failed to convert data to 'utf8' encoded string' with RNFetchBlob.fs.readStream

### ISSUE THREAD

**Issue by Kenny806** [wkh237/react-native-fetch-blob#593]
Hi! I am trying to read a CSV file with `RNFetchBlob.fs.readStream`, but keep getting this error:

```
Failed to convert data to 'utf8' encoded string, this might due to the source data is not able to convert using this encoding. source = *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]
```
Curious thing is, that when I use `readFile()` instead of `readStream()`, the operation runs just fine and I get a result. The problem is, I would like to use `readStream()` because some of the files I need to read are too big to be efficiently handled by `readFile()`.

Also, other files in the same directory and with the same format can be processed with both `readFile()` and `readStream()`.

Here is my code:

```
readFiles(path) {
    let data ='';
    RNFetchBlob.fs.readStream(`${path}/trips.txt`, 'utf8')
    .then((ifstream) => {
      ifstream.open();
      ifstream.onData((chunk) => {
        data += chunk;
      });
      ifstream.onError((err) => {
        console.log(err);
      });
      ifstream.onEnd(() => {
        this.processCSVData(data);
      });
    }).catch((err) => {
      console.log(err);
    });
  }
```
This is the part that works on some files and does not work on others (for example the [trips.txt](https://github.com/wkh237/react-native-fetch-blob/files/1459305/trips.txt)). I have not found anything that makes the files that work different from the files that don't. Worth noting is, that it always crashes inside the `readStream()` call, without ever getting to any of the callbacks.

What always works (except for very large files, obviously :-) ), is this:

```
RNFetchBlob.fs.readFile(`${path}/trips.txt`, 'utf8').then((contents) => {
  this.processCSVData(contents);
});
```
my RN and RNFetchBlob versions:
```
"react-native": "0.50.1",
 "react-native-fetch-blob": "^0.10.8",
```

Any help is appreciated.

Thanks!

**Comment by sumitnitsurat** [comment]
hey have you resolved this issue . I am facing the same issue.

**Comment by Kenny806** [comment]
No, not really, I had to use a workaround. If the error occurs, I use `RNFetchBlob.fs.readFile` to read the file's contents. Luckily for me, the error only occurs on files that are not too big to read with `readFile` (I have a more or less consistent dataset, and it's always the same two files that produce this error, even if the content changes). If the file is so big you need 'readStream', then you're out of luck I'm afraid.

**Comment by sumitnitsurat** [comment]
hey Thanks for the reply. Ill apply the same. Can you confirm me one more thing? I have xls file with 1000 rows and 10 column. RNFetchBlob.fs.readFile would work fine with it ? whats the limit of big for readFile().

**Comment by Kenny806** [comment]
I am processing a csv with ~70k rows and 6 columns, so you should be fine. There is no "limit" in the real sense of the word, the processing time just increases rapidly and at some point it gets useless. Just try it :-)

**Comment by sumitnitsurat** [comment]
Ahh..okay Thanks for the info. One last thing have you tried reading the .xls or .xlsx file. I am able to read .csv file but not .xls or .xlsx file.
